### PR TITLE
Deprecate Source.combine instance method (#22452)

### PR DESCRIPTION
**Pull request by nadavwr**

### CODE REVIEW

**Comment by ktoso** [comment]
Refs #22452 (reference has to be in comment, not only commit title)

Thanks :)

**Comment by ktoso** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala [hunk 1]
```diff
@@ -162,8 +162,9 @@ final class Source[+Out, +Mat](
   def asJava: javadsl.Source[Out, Mat] = new javadsl.Source(this)
 
   /**
-   * Combines several sources with fun-in strategy like `Merge` or `Concat` and returns `Source`.
+   * Combines several sources with fan-in strategy like `Merge` or `Concat` and returns `Source`.
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: Funny typo, thanks for spotting

**Review by ktoso — COMMENTED** [review]
LGTM, thanks

**Review by ktoso — APPROVED** [review]

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Pull request merged**